mlrun/feature_store/retrieval/base.py

```diff
@@ -240,14 +268,20 @@ def _generate_vector(
             # update alias according to the unique column name
             new_columns = []
             if not self._drop_indexes:
-                new_columns.extend([(ind, ind) for ind in fs_entities])
+                new_columns.extend([(ind, ind) for ind in fs_entities_and_timestamp])
             for column, alias in columns:
                 if column in rename_col_dict:
                     new_columns.append((rename_col_dict[column], alias or column))
                 else:
                     new_columns.append((column, alias))
             self._update_alias(dictionary={name: alias for name, alias in new_columns})
 
+        # None of the feature sets was filtered as required
+        if filtered and (start_time or end_time):
+            raise mlrun.errors.MLRunRuntimeError(
+                "start_time and end_time can only be provided in conjunction with "
+                "a timestamp column, or when the feature_sets has a timestamp_key"
```

Comment:
Do they all have feature sets to have a timestamp key, or would it be enough for just one of them to have it? (error should be clear about this)